src/components/connection_handler/src/connection_handler_impl.cc

```diff
@@ -403,22 +406,24 @@ void ConnectionHandlerImpl::OnSessionStartedCallback(
 #endif  // ENABLE_SECURITY
                         << " service " << static_cast<int>(service_type)
                         << " for session " << static_cast<int>(session_id));
-      NotifySessionStartedFailure(connection_handle, session_id, is_protected);
+      protocol_handler_->NotifySessionStarted(context, rejected_params);
       return;
     }
-    new_session_id = session_id;
-    hash_id = protocol_handler::HASH_ID_NOT_SUPPORTED;
+    context.new_session_id_ = session_id;
+    context.hash_id_ = protocol_handler::HASH_ID_NOT_SUPPORTED;
   }
   sync_primitives::AutoReadLock read_lock(connection_handler_observer_lock_);
   if (connection_handler_observer_) {
-    const uint32_t session_key = KeyFromPair(connection_handle, new_session_id);
+    const uint32_t session_key =
+        KeyFromPair(connection_handle, context.new_session_id_);
+
+    uint32_t app_id = 0;
+    GetDataOnSessionKey(session_key, &app_id, NULL, NULL);
```

Comment:
The build fails on this line in the new version of develop, 

```
/home/jacob/sdl_core_master/src/components/connection_handler/src/connection_handler_impl.cc: In member function ‘virtual void connection_handler::ConnectionHandlerImpl::OnSessionStartedCallback(transport_manager::ConnectionUID, uint8_t, const protocol_handler::ServiceType&, bool, const BsonObject*)’:
/home/jacob/sdl_core_master/src/components/connection_handler/src/connection_handler_impl.cc:471:57: error: call of overloaded ‘GetDataOnSessionKey(const uint32_t&, uint32_t*, NULL, NULL)’ is ambiguous
     GetDataOnSessionKey(session_key, &app_id, NULL, NULL);
                                                         ^
In file included from /home/jacob/sdl_core_master/src/components/connection_handler/src/connection_handler_impl.cc:39:0:
/home/jacob/sdl_core_master/src/components/connection_handler/include/connection_handler/connection_handler_impl.h:500:11: note: candidate: virtual int32_t connection_handler::ConnectionHandlerImpl::GetDataOnSessionKey(uint32_t, uint32_t*, std::list<int>*, connection_handler::DeviceHandle*) const
   int32_t GetDataOnSessionKey(
           ^~~~~~~~~~~~~~~~~~~
/home/jacob/sdl_core_master/src/components/connection_handler/include/connection_handler/connection_handler_impl.h:515:22: note: candidate: virtual int32_t connection_handler::ConnectionHandlerImpl::GetDataOnSessionKey(uint32_t, uint32_t*, std::list<int>*, uint32_t*) const
   DEPRECATED int32_t GetDataOnSessionKey(uint32_t key,
```

casting the last argument to `DeviceHandle*` should do the trick.

Reply:
@jacobkeeler yes, there was a build issue after rebase. Fixed in 7898292b5de8d80354ab3b9fecacbadcf9d41c61
Now build is OK.